### PR TITLE
Fixes for #19545

### PR DESCRIPTION
**Pull request by omus**
Addressing @tkelman's comments.

### CODE REVIEW

**Comment by tkelman** [comment]
Cool. Shall we squash-merge this into `sh/fast_date_parsing` since these commits are so tiny? CI doesn't run on PR's-to-PR's with our current configuration until after they are merged.

File: base/dates/parse.jl [hunk 1]
```diff
@@ -110,59 +126,58 @@ function Base.parse(::Type{DateTime}, s::AbstractString, df::DateFormat{Symbol("
     th = tm = ts = tms = Int64(0)
 
     nv, i = tryparsenext_base10(s,i,end_pos,1)
-    dy = isnull(nv) ? (@goto error) : get(nv)
+    dy = isnull(nv) ? (@goto error) : nv.value
```
Comment (tkelman) [inline base/dates/parse.jl]: not sure whether `unsafe_get` might be preferred ?

Comment (omus) [inline base/dates/parse.jl]: I was unaware of `unsafe_get`. I'll update to use it and make sure there isn't any performance hit.

Comment (tkelman) [inline base/dates/parse.jl]: Ah yeah it's exported and documented. Longer to type but presumably more likely to survive any modification to representation of `Nullable`s.

**Comment by omus** [comment]
Feel free to squash the commits. Maybe keep the NTuple changes as a separate commit.

**Comment by tkelman** [comment]
Like that? Wonder why github still shows a nonzero-line diff here.

**Comment by tkelman** [comment]
the meanings of `tryparsenext`, `fixwidth`, and `reorder_args` could still use some explanatory comments, but are those maybe parts of the code that @shashi wrote?

**Comment by shashi** [comment]
I'll write comments for those.

**Pull request closed without merging**